thorlcr/graph/thgraph.hpp

```diff
@@ -208,16 +209,24 @@ class CFileUsageEntry : public CInterface
     void decUsage() { --usage; }
 
     const char *queryFindString() const { return name; }
+    inline void setSize(offset_t _size) { size = _size; }
+    inline offset_t getSize() const { return size; }
 };
 
 typedef IIteratorOf<CFileUsageEntry> IFileUsageIterator;
 
 interface IGraphTempHandler : extends IInterface
 {
-    virtual void registerFile(const char *name, graph_id graphId, unsigned usageCount, bool temp, WUFileKind fileKind=WUFileStandard, StringArray *clusters=NULL) = 0;
+    virtual CFileUsageEntry * registerFile(const char *name, graph_id graphId, unsigned usageCount, bool temp, WUFileKind fileKind=WUFileStandard, StringArray *clusters=NULL) = 0;
     virtual void deregisterFile(const char *name, bool kept=false) = 0;
     virtual void clearTemps() = 0;
     virtual IFileUsageIterator *getIterator() = 0;
+    virtual void serializeUsageStats(MemoryBuffer &mb, graph_id gid) = 0;
+    static void serializeNullUsageStats(MemoryBuffer &mb)
+    {
+        mb.append(0);
```

Comment:
I don't think this is correct.  It will append an int (4 bytes), instead of an offset_t (8 bytes)

Reply:
you're right, should be changed to something like:
```
mb.append((offset_t)0);
mb.append((offset_t)0);
```